indigo_social/views.py

```diff
@@ -292,3 +293,17 @@ class UserPopupView(AbstractAuthedIndigoView, DetailView):
     slug_url_kwarg = 'username'
     template_name = 'indigo_social/user_popup.html'
     queryset = User.objects
+
+
+class UserProfilePhotoView(View):
+    def get(self, request, **kwargs):
+        username = kwargs['username']
+        nonce = kwargs['nonce']
+
+        user = User.objects.filter(username=username).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+
+        response = HttpResponse(user_profile.profile_photo.read())
```

Comment:
The photo might not exist, redirect to the static image in that case.

Reply:
The physical photo? We already redirect to the static photo when getting the photo_url in the model

Reply:
There's no guarantee this object will exist in the db by the time the request comes here. A user could load a page from their cache which is old, etc.